### PR TITLE
Schema config resource

### PR DESCRIPTION
**Pull request by beevital**
New resource added for managing connector schema config

### CODE REVIEW

**Comment by beevital** [comment]
@5tran-alexil please take a look at .md file

File: fivetran/resource_connector_schema_config.go [hunk 1]
```diff
@@ -0,0 +1,810 @@
...
+			"connector_id":           {Type: schema.TypeString, Required: true, ForceNew: true},
+			"schema_change_handling": {Type: schema.TypeString, Required: true},
+			"schema":                 resourceSchemaConfigSchema(),
+		},
+	}
+}
+
+func resourceSchemaConfigSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
```
Comment (Fivetran-RomanKolesnikov) [inline fivetran/resource_connector_schema_config.go:35]: Formatting can be as in the following schemas (type, optional, set in one line)

File: fivetran/resource_connector_schema_config.go [hunk 2]
```diff
@@ -0,0 +1,810 @@
...
+				"name":          {Type: schema.TypeString, Required: true},
+				"enabled":       {Type: schema.TypeString, Optional: true, Default: "true"},
+				"patch_allowed": {Type: schema.TypeString, Computed: true},
+				"hashed":        {Type: schema.TypeString, Optional: true, Default: "false"},
+			},
+		},
+	}
+}
+
+// CREATE
```
Comment (Fivetran-RomanKolesnikov) [inline fivetran/resource_connector_schema_config.go:74]: Do we need this comments?

**Review by Fivetran-RomanKolesnikov — APPROVED** [review]
Great work, thank you!

File: fivetran/resource_connector_schema_config.go [hunk 3]
```diff
@@ -0,0 +1,810 @@
...
+		if hashed, ok := cmap["hashed"].(string); ok && hashed != "" {
+			rcolumn["hashed"] = hashed
+		}
+
+		result[cname] = rcolumn
+	}
+	return result
+}
+
+func flatternConfig(config map[string]interface{}) map[string]interface{} {
```
Comment (dkanareykin) [inline fivetran/resource_connector_schema_config.go:509]: Maybe flatten?
```suggestion
func flattenConfig(config map[string]interface{}) map[string]interface{} {
```

**Review by dkanareykin — APPROVED** [review]
Looks good.

From my point of view, the algorithm here is quite complex and it would be great if it is documents step-by-step.

There are several functions that perform almost the same actions for different entities (for schemas and tables), it's possible to make a universal ones, but it will make the code even more difficult to read.

Probably we can try major refactoring this code when (if) we have time.

For now it's possible to consider introduction of a new type
```go
type iMap map[string]interface{}
```
because it is widely used in the code.

File: fivetran/resource_connector_schema_config.go (hunk 2 above)
Comment (dkanareykin) [reply fivetran/resource_connector_schema_config.go:74]: Comments are good :)

**Review by 5tran-alexil — COMMENTED** [review]
Made some edits.
@beevital Since I have edited some entity names and the relevant code blocks, please check for technical accuracy.

File: fivetran/resource_connector_schema_config.go [hunk 4]
```diff
@@ -0,0 +1,810 @@
...
+				"name":    {Type: schema.TypeString, Required: true},
+				"enabled": {Type: schema.TypeString, Optional: true, Default: "true"},
+				"table":   resourceSchemaConfigTable(),
+			},
+		},
+	}
+}
+
+func resourceSchemaConfigTable() *schema.Schema {
+	return &schema.Schema{Type: schema.TypeSet, Optional: true, Set: resourceTableConfigHash,
```
[the remark this answers is not in the document]
Comment (beevital) [reply fivetran/resource_connector_schema_config.go:46]: I don't think it's valuable, in other resources we define it inline. Let it be as it is now

**Pull request merged**